kvdb-lmdb/benches/kvdb-lmdb.rs

```diff
@@ -0,0 +1,125 @@
+use kvdb::{DBTransaction, NumEntries};
+use elapsed::measure_time;
+use lmdb::{EnvironmentFlags, DatabaseFlags, WriteFlags};
+
+fn randbytes(n: usize) -> Vec<u8> {
+    let mut buf = vec![0; n];
+    thread_rng().fill(&mut buf[..]);
+    buf
+}
+
+// Creates a database populated with sequential numbers as keys, and random bytes as values. Used
+// only to provide other tests with a database with data in it to emulate a more realistic workload
+// in the actual tests. The write settings here are highly unsafe and are not used in the actual
+// tests.
+fn create_benchmark_db() -> String {
+    use std::{fs, path};
+    let size = 10_000_000;
+    let batch_size = 1_000_000;
+    let path = format!("./benches/dbs/bench-lmdb-full-{}", size);
+    let meta = fs::metadata(path::Path::new(&format!("{}/data.mdb", path)));
+    // These settings are **NOT** ok for any other use than throw-away data
+    let db_cfg = DatabaseConfig::new(1)
+        .with_env_flags(EnvironmentFlags::NO_META_SYNC | EnvironmentFlags::NO_LOCK | EnvironmentFlags::NO_SYNC)
+        .with_db_flags(DatabaseFlags::INTEGER_KEY)
+        .with_write_flags(WriteFlags::APPEND);
+    let db = Database::open(&db_cfg, &path.clone()).unwrap();
+
+    let db_ok = if meta.is_err() {
+        false
+    } else {
+        if let Ok(entries) = db.num_entries(0) {
+            println!("DB has {} entries", entries);
+            if entries == size { true } else { false }
+        } else {
+            false
+        }
+    };
+
+    // Try to avoid re-creating the test DB for every test run.
+    if !db_ok {
+        println!("Creating benchmark DB");
+        let (elapsed, _) = measure_time(|| {
+            let batches = size / batch_size;
+            for b in 0..batches {
+                println!("Batch: {}/{} – inserting indices {} –> {}", b, batches, b*batch_size, (b+1)*batch_size);
+                let mut tr = DBTransaction::with_capacity(batch_size);
+                for i in b*batch_size..(b+1)*batch_size {
+                    let key = b*batch_size + i;
+                    let slice = &unsafe { std::mem::transmute::<usize, [u8; 8]>(key) };
+                    let v = randbytes(200); // TODO: need the distribution of payload sizes; match that to a distribution from `rand` and generate rand bytes accordingly?
+                    tr.put(None, slice.as_ref(), &v);
+                }
+                db.write(tr).unwrap();
+            }
+        });
+        println!("Created benchmark DB in {}", elapsed);
+    }
+    path
+}
+
+fn write_to_empty_db(c: &mut Criterion) {
+    let path = "./benches/dbs/bench-lmdb-empty";
+    let path2 = path.clone();
+    c.bench(
+        "empty DB, 32 byte keys",
+        ParameterizedBenchmark::new(
+            "payload size",
+            move |b, payload_size| {
+                let cfg = DatabaseConfig::new(1u32);
+                let db = Database::open(&cfg, path).unwrap();
+                let v = randbytes(*payload_size);
```

Comment:
Instead of using one value of `payload_size`, it would make sense to insert batch of `payload_size / 200` values of size 200 (and probably rename `payload_size` to `batch_size`).

Reply:
Definitely need something better than a hard coded `200` here, but I suspect the payload size distribution is fairly important and I want to collect more data on it. So I'll return to this.

(I also think this "empty DB" test is fairly useless, will prolly remove it entirely)